compiler/src/dotty/tools/backend/jvm/GenBCode.scala

```diff
@@ -558,18 +560,29 @@ class GenBCodePipeline(val int: DottyBackendInterface, val primitives: DottyPrim
     *    (c) dequeue one at a time from queue-2, convert it to byte-array,    place in queue-3
     *    (d) serialize to disk by draining queue-3.
     */
-  private def buildAndSendToDisk(needsOutFolder: Boolean) = {
-
-    feedPipeline1()
-    // val genStart = Statistics.startTimer(BackendStats.bcodeGenStat)
-    (new Worker1(needsOutFolder)).run()
-    // Statistics.stopTimer(BackendStats.bcodeGenStat, genStart)
-
-    (new Worker2).run()
-
-    // val writeStart = Statistics.startTimer(BackendStats.bcodeWriteTimer)
-    drainQ3()
-    // Statistics.stopTimer(BackendStats.bcodeWriteTimer, writeStart)
+  private def buildAndSendToDisk(needsOutFolder: Boolean)(using Context) = {
+    try
+      feedPipeline1()
+      // val genStart = Statistics.startTimer(BackendStats.bcodeGenStat)
+      (new Worker1(needsOutFolder)).run()
+      // Statistics.stopTimer(BackendStats.bcodeGenStat, genStart)
+
+      (new Worker2).run()
+
+      // val writeStart = Statistics.startTimer(BackendStats.bcodeWriteTimer)
+      drainQ3()
+      // Statistics.stopTimer(BackendStats.bcodeWriteTimer, writeStart)
+    catch
+      case e: MethodTooLargeException =>
+        val method = 
+          s"${e.getClassName.replaceAll("/", ".")}.${e.getMethodName}"
+        val msg = 
+          s"Generated bytecode for method '$method' is too large. Size: ${e.getCodeSize} bytes. Limit is 64KB" 
+        report.error(msg)
+      case e: ClassTooLargeException =>
+        val msg = 
+          s"Class '${e.getClassName.replaceAll("/", ".")}' is too large. Constant pool size: ${e.getConstantPoolCount}. Limit is 64K"
```

Comment:
```suggestion
          s"Class '${e.getClassName.replaceAll("/", ".")}' is too large. Constant pool size: ${e.getConstantPoolCount}. Limit is 64KB"
```

Reply:
Actually I did a check and in this case the unit of measure is not bytes but just an amount of entries in constant pool. 
https://github.com/scala/scala-asm/blob/8ff30b68d52ec42a63677a15b30195d7973012ef/src/main/java/scala/tools/asm/ClassWriter.java#L596-L600